department-of-veterans-affairs/caseflow
# CASEFLOW-96 and -282 bugfixes

Fixes some UI bugs with CASEFLOW-96 and CASEFLOW-282

### Description
I prematurely merged my previous PR&mdash;it had an approval but I hadn't demoed it to Ann-Marie. D'oh!

![doh](https://media.giphy.com/media/P8WZZ0NYdbXAA/giphy.gif)

She found a couple things that needed to be fixed, which this PR does:
* Filling in dates for judgement and mandate dates, but then hiding the fields, caused the dates to get saved.
* A copy change.
* I also discovered that selecting MDR, then changing the type to a death dismissal would cause the "Do you have a mandate?" question to not be displayed.

### Acceptance Criteria
Bugfix: see testing plan.

### Testing Plan

- Enter the "Add CAVC Remand" form
- Select, but do not submit, "Remand" for "How are you proceeding?", and MDR for "What type of remand is this?"
- Change "How are you proceeding?" to "Death Dismissal"
- Confirm that "Are judgement and mandate dates provided?" is shown. (It was previously not if MDR had been selected even though it no longer applied.)
- Leave "Yes" checked on "Are judgement and mandate dates provided?"
- Input dates for all 3 date fields.
- Select "No" for "Are judgement and mandate dates provided?"
- Fill in the other fields as needed to pass validation( values are unimportant) and submit it.
- Scroll to the CAVC Remand section on the page and confirm that only the decision date is shown. (Previously, judgement and mandate dates showed if they had been filled out but then the form was changed to indicate they were unavailable.)

- Enter the Add CAVC Remand form again
- Enter an MDR, filling in the form normally and populating fields as needed to pass validation
- Confirm that the displayed success message says "This has generated an MDR task in your team's **Assigned** tab and will return to the Unassigned tab after 90 days from the court decision date" (bold text only for context here; it previously read "On Hold" which was incorrect)

#### Getting to the Add CAVC Remand form

The lazy way: search for case 500000000 (eight zeroes) and select something in "Dispatched state"

The Rails console way:

```ruby
RequestStore[:current_user] = User.system_user

original_appeal = FactoryBot.create(:appeal, 
                           :dispatched, 
                           associated_judge: JudgeTeam.first.judge, 
                           associated_attorney: JudgeTeam.first.attorneys.first);
FactoryBot.create_list(:request_issue, 2,                                                                                  
            :rating,                                                                                            
            :with_rating_decision_issue,                                                                           
            decision_review: original_appeal,                                                                               
            veteran_participant_id: Veteran.first.participant_id,                                                        
            contested_issue_description: "issue_description",                                                              
            notes: "issue_notes");

original_appeal.uuid
```

And navigate to /queue/appeals/UUID

In either case, the "+ Add CAVC Remand" will be visible at the top of the page.

### User Facing Changes

Only one makes sense to screenshot; the others are bugfixes where fields behaved inappropriately in certain cases:

![Screen Shot 2021-02-09 at 1 49 08 PM](https://user-images.githubusercontent.com/314663/107412412-a1cfd100-6add-11eb-8c67-466d89577f4a.png)
